src/label-helpers.ts

```diff
@@ -22,37 +25,46 @@ function getTextContent(node) {
     .join('')
 }
 
-function getLabelContent(node) {
+function getLabelContent(node: Node | Element | HTMLInputElement) {
   let textContent
-  if (node.tagName.toLowerCase() === 'label') {
+  if ('tagName' in node && node.tagName.toLowerCase() === 'label') {
     textContent = getTextContent(node)
+  } else if ('value' in node) {
+    textContent = node.value
   } else {
-    textContent = node.value || node.textContent
+    textContent = node.textContent
```

Comment:
Could this work?

```suggestion
    textContent = node?.value ?? node.textContent
```

Or even

```suggestion
    textContent = node.value ?? node.textContent
```